### ISSUE TITLE
How do I get the citation key in postscript?

### ISSUE THREAD

**Issue by coldacid** [retorquere/zotero-better-bibtex#933]
Is it possible to get access to the generated or pinned citekey value in the postscript JS? I'd like to add an explicit key string to BibTeX exports since the citeproc I'm using doesn't think to use the regular key when generating citation text.

**Comment by retorquere** [comment]
You can get it from `item.citekey`. But I don't understand how you're trying to use it -- citeproc doesn't do BibTeX.

**Comment by coldacid** [comment]
pandoc-citeproc can take in Bib(La)TeX as one of its input formats for handling citations in Markdown documents. However, I've found a slightly better workflow that lets me use CSL YAML with it instead.

**Comment by retorquere** [comment]
If you export using Better CSL YAML, you'll get the citekeys as the reference IDs.

**Comment by coldacid** [comment]
@retorquere yeah, that's what I did, although I still need `citation-label` explicitly set. Fortunately a simple sed command in my toolchain takes care of that.

**Comment by retorquere** [comment]
You can set the citation-label in the postscript. Postscripts work for the CSL exports too.

**Comment by retorquere** [comment]
On 5.0.105 You'd use a postscript such as

```
if (Translator.BetterCSL) { // covers both Translator.BetterCSLJSON and Translator.BetterCSLYAML
  reference['citation-label'] = item.citekey;
}
```

(on pre-.105, you'd have to test against Translator.BetterCSLJSON and Translator.BetterCSLYAML separately)

**Comment by coldacid** [comment]
@retorquere having done that, I don't get `citation-label` in the exported YAML even with .105. For now I'm going to stick with my sed regex to duplicate `id` as `citation-label`.

**Comment by blip-bloop** [comment]
:robot: this is your friendly neighborhood build bot announcing [test build 6016](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-5.0.105.6013.issue-933.xpi) ("fixes #933").

**Comment by retorquere** [comment]
My bad, I had forgotten to merge a branch. The postscript works on 6016.
